File: staffapp/src/main/java/com/example/staffapp/UiLabels.kt
package com.example.staffapp

object UiLabels {
    private val roleMap = mapOf(
        "ROLE_TRAINER" to "Тренер",
        "ROLE_MANAGER" to "Менеджер",
        "ROLE_FINANCE" to "Финансы",
        "ROLE_VIEWER" to "Наблюдатель",
        "ROLE_SUPPORT" to "Поддержка",
        "ROLE_ADMIN" to "Администратор",
        "ROLE_SUPER_ADMIN" to "Суперадминистратор",
    )

    private val sectionMap = mapOf(
        "home" to "Главная",
        "profile" to "Профиль",
        "dashboard" to "Дашборд",
        "tasks" to "Задачи",
        "clients" to "Клиенты",
        "schedule" to "Расписание",
        "bookings" to "Записи",
        "subscriptions" to "Абонементы",
        "visits" to "Посещения",
        "analytics" to "Аналитика",
        "finance" to "Финансы",
        "trainers" to "Тренеры",
        "documents" to "Документы",
        "crm_staff" to "Персонал CRM",
        "app_support" to "Обращения",
        "leads" to "Лиды",
        "deals" to "Сделки",
        "sales" to "Продажи",
        "services" to "Услуги",
        "visits" to "Посещения",
        "analytics" to "Аналитика",
        "promocodes" to "Промокоды",
        "promotions" to "Акции",
        "tags" to "Теги",
        "comments" to "Комментарии",
        "settings" to "Настройки",
        "warehouse" to "Склад",
        "cashdesk" to "Касса",
        "mobileapps" to "Мобильные приложения",
        "messengers" to "Мессенджеры",
        "calls" to "Звонки",
        "selfservice" to "Самообслуживание",
        "franchise" to "Франшиза",
    )

    fun sectionTitle(key: String): String = sectionMap[key] ?: key.replaceFirstChar { it.uppercase() }
    fun roleTitle(code: String): String = roleMap[code] ?: code

    fun metricTitle(key: String): String = when (key) {
        "clients" -> "Клиенты"
        "bookings" -> "Записи"
        "subscriptions" -> "Абонементы"
        "tasks_open" -> "Открытые задачи"
        "tasks_done" -> "Закрытые задачи"
        "all_clients" -> "Всего клиентов"
        "all_bookings" -> "Всего записей"
        "all_subscriptions" -> "Всего абонементов"
        "trainers_total" -> "Всего тренеров"
        "documents_total" -> "Всего документов"
        "staff_total" -> "Всего сотрудников"
        "trainers" -> "Тренеры"
        "documents" -> "Документы"
        "staff" -> "Сотрудники"
        "schedule_today" -> "Тренировок сегодня"
        "schedule_tomorrow" -> "Тренировок завтра"
        "schedule_week" -> "Тренировок на этой неделе"
        "schedule_next_week" -> "Тренировок на следующей неделе"
        "items_total" -> "Элементов"
        "support_new" -> "Новых обращений"
        "notifications_unread" -> "Непрочитанных уведомлений"
        "leads_new" -> "Новых лидов"
        "leads_converted" -> "Конвертированных лидов"
        "deals_total" -> "Сделок"
        "sales_total" -> "Продаж"
        "visits_today" -> "Проходов сегодня"
        "finance_income_month" -> "Доход за месяц"
        "finance_expense_month" -> "Расход за месяц"
        "products_total" -> "Услуг в каталоге"
        "promocodes_total" -> "Промокодов"
        "promotions_total" -> "Акций"
        "tags_total" -> "Тегов"
        "comments_total" -> "Комментариев"
        else -> key
    }

    fun ticketCategory(category: String): String = when (category) {
        "question" -> "Вопрос"
        "complaint" -> "Жалоба"
        "suggestion" -> "Предложение"
        "technical" -> "Техническая проблема"
        "billing" -> "Оплата / абонемент"
        else -> "Другое"
    }

    fun ticketStatus(status: String): String = when (status) {
        "new" -> "Новое"
        "in_progress" -> "В работе"
        "done" -> "Закрыто"
        else -> status
    }

    fun trainingType(type: String): String = when (type) {
        "personal" -> "Персональная"
        "group" -> "Групповая"
        "extra" -> "Услуга"
        else -> "Занятие"
    }
}

File: MatskocaAdmin/MatskocaAdmin/Labels/UiLabels.swift
import Foundation

enum UiLabels {
    private static let roleMap: [String: String] = [
        "ROLE_TRAINER": "Тренер",
        "ROLE_MANAGER": "Менеджер",
        "ROLE_FINANCE": "Финансы",
        "ROLE_VIEWER": "Наблюдатель",
        "ROLE_SUPPORT": "Поддержка",
        "ROLE_ADMIN": "Администратор",
        "ROLE_SUPER_ADMIN": "Суперадминистратор",
    ]

    private static let sectionMap: [String: String] = [
        "home": "Главная",
        "profile": "Профиль",
        "dashboard": "Дашборд",
        "tasks": "Задачи",
        "clients": "Клиенты",
        "schedule": "Расписание",
        "bookings": "Записи",
        "subscriptions": "Абонементы",
        "visits": "Посещения",
        "analytics": "Аналитика",
        "finance": "Финансы",
        "trainers": "Тренеры",
        "documents": "Документы",
        "crm_staff": "Персонал CRM",
        "app_support": "Обращения",
        "leads": "Лиды",
        "deals": "Сделки",
        "sales": "Продажи",
        "services": "Услуги",
        "promocodes": "Промокоды",
        "promotions": "Акции",
        "tags": "Теги",
        "comments": "Комментарии",
        "settings": "Настройки",
        "warehouse": "Склад",
        "cashdesk": "Касса",
        "mobileapps": "Мобильные приложения",
        "messengers": "Мессенджеры",
        "calls": "Звонки",
        "selfservice": "Самообслуживание",
        "franchise": "Франшиза",
    ]

    static func sectionTitle(_ key: String) -> String {
        sectionMap[key] ?? key.prefix(1).uppercased() + key.dropFirst()
    }

    static func roleTitle(_ code: String) -> String {
        roleMap[code] ?? code
    }

    static func metricTitle(_ key: String) -> String {
        switch key {
        case "clients": return "Клиенты"
        case "bookings": return "Записи"
        case "subscriptions": return "Абонементы"
        case "tasks_open": return "Открытые задачи"
        case "tasks_done": return "Закрытые задачи"
        case "all_clients": return "Всего клиентов"
        case "all_bookings": return "Всего записей"
        case "all_subscriptions": return "Всего абонементов"
        case "trainers_total": return "Всего тренеров"
        case "documents_total": return "Всего документов"
        case "staff_total": return "Всего сотрудников"
        case "trainers": return "Тренеры"
        case "documents": return "Документы"
        case "staff": return "Сотрудники"
        case "schedule_today": return "Тренировок сегодня"
        case "schedule_tomorrow": return "Тренировок завтра"
        case "schedule_week": return "Тренировок на этой неделе"
        case "schedule_next_week": return "Тренировок на следующей неделе"
        case "items_total": return "Элементов"
        case "support_new": return "Новых обращений"
        case "notifications_unread": return "Непрочитанных уведомлений"
        case "leads_new": return "Новых лидов"
        case "leads_converted": return "Конвертированных лидов"
        case "deals_total": return "Сделок"
        case "sales_total": return "Продаж"
        case "visits_today": return "Проходов сегодня"
        case "finance_income_month": return "Доход за месяц"
        case "finance_expense_month": return "Расход за месяц"
        case "products_total": return "Услуг в каталоге"
        case "promocodes_total": return "Промокодов"
        case "promotions_total": return "Акций"
        case "tags_total": return "Тегов"
        case "comments_total": return "Комментариев"
        default: return key
        }
    }

    static func ticketCategory(_ category: String) -> String {
        switch category {
        case "question": return "Вопрос"
        case "complaint": return "Жалоба"
        case "suggestion": return "Предложение"
        case "technical": return "Техническая проблема"
        case "billing": return "Оплата / абонемент"
        default: return "Другое"
        }
    }

    static func ticketStatus(_ status: String) -> String {
        switch status {
        case "new": return "Новое"
        case "in_progress": return "В работе"
        case "done": return "Закрыто"
        default: return status
        }
    }

    static func trainingType(_ type: String) -> String {
        switch type {
        case "personal": return "Персональная"
        case "group": return "Групповая"
        case "extra": return "Услуга"
        default: return "Занятие"
        }
    }
}
